Minor bug fixes

Jaserii opened the pull request:
-Fixed URL when clicking the share button (Still needs to be updated to support the short links)
-Updated the reset password functionality to use password encoding
-Updated PasswordResetController to utilize a try-catch in case the program's SMTP configuration is invalid
-Cleaned up code in UserServiceImpl and PasswordResetController

Jaserii committed:
Fix url in social sharing function

Minor fix. The bug was that the code was treating the summarized content as the url.

Jaserii committed:
Update email reset to use password encoding

Jaserii committed:
Update pw-reset controller

Now uses the existing "loadUserByUsername" method found in UserServiceImpl.java.

Jaserii committed:
Update pw-reset controller with try-catch

Added another try-catch to the email functionality in case the SMTP configuration in env.properties stops working and it doesn't break the runtime.

17YuvrajSehgal commented on src/main/java/com/websummarizer/Web/Summarizer/controller/PasswordResetController.java:
```diff
@@ -119,10 +147,9 @@ public String reset(
         if (temp != null) {
             temp.setPassword(password);
-            userService.setPassword(temp);
+            authenticationService.registerUser(temp);
```
instead of making an authentication service object, try to use authentication controller, since no user should be able to communicate with services.

Jaserii replied:
I've committed the change to github. What about my usage of the user service object? Is that fine to keep as is?

Jaserii committed:
Use AuthenticationController

Jaserii committed:
Remove unnecessary code

Removed unused/duplicative methods from UserRepo and UserServiceImpl. Updated comments

Jaserii committed:
Removed unused class

Removed PasswordResetHTTPStatus because it was initially only used for unit testing but then I ended up not even using it
